Author: Fabian Greffrath <fabian+debian@greffrath.com>
Description: Allocate a whole 16 MB at once instead of iterating in
 128 kB steps. Thanks again, Jakub Wilk.
Bug-Debian: https://bugs.debian.org/775508

--- a/source/base/all/memory/lin.c
+++ b/source/base/all/memory/lin.c
@@ -22,6 +22,7 @@ struct 	  rusage ru;
 
   Buf = NULL;
 
+#if 0
   while (Size < BASE_MEMORY_EXTERN_MaxMemoryRequirement()
          && (NewBuf = (char*) realloc(Buf, Size + STEP)))
   {
@@ -48,6 +49,8 @@ struct 	  rusage ru;
 
   free(Buf);
   BASE_MEMORY.MaxAllocate  = Size;
+#endif
+  BASE_MEMORY.MaxAllocate  = BASE_MEMORY_EXTERN_MaxMemoryRequirement();
 
   BASE_MEMORY_InitAlloc();
 }
